go_modules/lib/dependabot/go_modules/version.rb

```diff
@@ -39,6 +42,73 @@ def inspect # :nodoc:
       def to_s
         @version_string
       end
+
+      def <=>(other)
+        result = super(other)
+        return if result.nil?
+        return result unless result.zero?
+
+        other = self.class.new(other) unless other.is_a?(Version)
+        compare_prerelease(@prerelease || "", T.unsafe(other).prerelease || "")
```

Comment:
I think this is because of the signature on `def <=>` of `Dependabot::Version`. If this needs to be overridden in every `Dependabot::ECOSYSTEM::Version` class then it needs to be a generic. See https://sorbet.org/docs/override-checking#what-if-i-really-want-the-child-method-to-narrow-the-type

I'm not asking you to fix it, more of in case you're interested.